File: aab-hl-application/Assets/00 BasicCode/1 Script/Networking/EventHandling/JsonRpcEvent.cs
namespace MasterNetworking.EventHandling
{
    public enum JsonRpcEvent
    {
        EmulatedOptiTrackMessage,
        DeviceRegistration,
        OptiTrackMessageRequest,
        OptiTrackMessage,
        OpenConnections,
    }

    public static class JsonRpcEventUtils
    {
        public static string ToString(JsonRpcEvent eventName)
        {
            switch (eventName)
            {
                case JsonRpcEvent.EmulatedOptiTrackMessage: return "EmulatedOptiTrackMessage";
                case JsonRpcEvent.DeviceRegistration:       return "DeviceRegistration";
                case JsonRpcEvent.OptiTrackMessageRequest:  return "OptiTrackMessageRequest";
                case JsonRpcEvent.OptiTrackMessage:         return "OptiTrackMessage";
                case JsonRpcEvent.OpenConnections:          return "OpenConnections";
            }

            return null;
        }

        public static JsonRpcEvent? FromString(string eventName)
        {
            switch (eventName)
            {
                case "EmulatedOptiTrackMessage":    return JsonRpcEvent.EmulatedOptiTrackMessage;
                case "DeviceRegistration":          return JsonRpcEvent.DeviceRegistration;
                case "OptiTrackMessageRequest":     return JsonRpcEvent.OptiTrackMessageRequest;
                case "OptiTrackMessage":            return JsonRpcEvent.OptiTrackMessage;
                case "OpenConnections":             return JsonRpcEvent.OpenConnections;
            }

            return null;
        }
    }
}

File: aab-web-application/_js/networking/eventHandling/jsonRpcEvent.js

/**
 * Defines predefined json rpc event types.
 * @type {{OpenConnections: string, OptiTrackMessageRequest: string, DeviceRegistration: string, OptiTrackMessage: string, EmulatedOptiTrackMessage: string}}
 */
const JsonRpcEvent = {
    "EmulatedOptiTrackMessage" : "EmulatedOptiTrackMessage",
    "DeviceRegistration" : "DeviceRegistration",
    "OptiTrackMessageRequest" : "OptiTrackMessageRequest",
    "OptiTrackMessage" : "OptiTrackMessage",
    "OpenConnections" : "OpenConnections"
}

export {
    JsonRpcEvent
}